sidechain/block-composer/src/block_composer.rs

```diff
@@ -126,15 +129,31 @@ where
 			Error::Other(format!("Failed to encrypt state payload: {:?}", e).into())
 		})?;
 
+		let now = now_as_u64();
+		let layer_one_hash = latest_parentchain_header.hash();
+		let payload_hash = calculate_payload_hash(
+			now,
+			layer_one_hash,
+			author_public,
+			top_call_hashes.clone(),
+			payload.clone(),
```

Comment:
Do we really need clones here? Can't we calculate the hash by passing in references?